multimodal/src/autogluon/multimodal/learners/base.py

```diff
@@ -0,0 +1,2473 @@
+                            ]
+                            if monitor_op(cand_score, best_score):
+                                # Add new ingredient
+                                ingredients.append(top_k_model_paths[i])
+                                best_score = cand_score
+                elif top_k_average_method == BEST:
+                    ingredients = [top_k_model_paths[0]]
+                else:
+                    raise ValueError(
+                        f"The key for 'optimization.top_k_average_method' is not supported. "
+                        f"We only support '{GREEDY_SOUP}', '{UNIFORM_SOUP}' and '{BEST}'. "
+                        f"The provided value is '{top_k_average_method}'."
+                    )
+        else:
+            # best_k_models is empty so we will manually save a checkpoint from the trainer
+            # and use it as the main ingredients
+            ingredients = [last_ckpt_path]
+            top_k_model_paths = []
+            # no checkpoints are available, do nothing
+            if not os.path.isfile(last_ckpt_path):
+                return
+
+        # Average all the ingredients
+        avg_state_dict = average_checkpoints(
+            checkpoint_paths=ingredients,
+        )
+        self._load_state_dict(
+            state_dict=avg_state_dict,
+            prefix=prefix,
+            strict=strict_loading,
+        )
+
+        if self._teacher_learner is not None:
+            avg_state_dict = self._replace_model_name_prefix(
+                state_dict=avg_state_dict,
+                old_prefix="student_model",
+                new_prefix="model",
+            )
+
+        if not standalone:
+            checkpoint = {"state_dict": avg_state_dict}
+        else:
+            if isinstance(strategy, DeepSpeedStrategy):
+                checkpoint = {
+                    "state_dict": {
+                        name.partition("module.")[2]: param
+                        for name, param in strategy.model._zero3_consolidated_16bit_state_dict().items()
+                    }
+                }
+            else:
+                checkpoint = {
+                    "state_dict": {"model." + name: param for name, param in self._model.state_dict().items()}
+                }
+
+        torch.save(checkpoint, os.path.join(save_path, MODEL_CHECKPOINT))
+
+        if clean_ckpts:
+            # clean old checkpoints + the intermediate files stored
+            for per_path in top_k_model_paths:
+                if os.path.isfile(per_path):
+                    os.remove(per_path)
+            # remove the yaml file after cleaning the checkpoints
+            if os.path.isfile(best_k_models_yaml_path):
+                os.remove(best_k_models_yaml_path)
+            # clean the last checkpoint
+            if os.path.isfile(last_ckpt_path):
+                os.remove(last_ckpt_path)
+
+    def prepare_for_deepspeed_offloading(self, strategy):
```

Comment:
This function is temporary and should be removed in future since it is mainly to support deepspeed offloading in prediction. It returns not only strategy, but also some optimization kwargs.